internal/firmware/info_fw.go

```diff
@@ -0,0 +1,57 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package firmware
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+)
+
+// fwInfoPath points to firmware metadata located in the
+// rootfs of a running system.
+var fwInfoPath string = "/termus/meta/termus.json"
+
+// fwInfo represents firmware metadata created during the firmware
+// build process.
+type fwInfo struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+	Summary string `json:"summary"`
+}
+
+// fwInfoFromRunning returns the firmware metadata of the running system
+// exported as an Info struct.
+func fwInfoFromRunning() (*Info, error) {
```

Comment:
Even though the "running firmware" is a good statement I don't think "Running" is the right term to use as if/when we introduce the non-running firmware the naming may not be as clean. May be "active" vs "inactive"?

Reply:
The challenge is that a firmware may be marked as active, but the reboot has not yet taken place, so its not running yet.

The other variant of this function will be called "metaInfoFromSlot()" to indicate indirect metadata extraction. So we will not need to have a "NotRunning...()".

I am going to keep this for now, until more complaints pile up.